src/panels/config/devices/ha-config-devices-dashboard.ts

```diff
@@ -130,25 +133,38 @@ export class HaConfigDeviceDashboard extends LitElement {
               direction: "asc",
               grows: true,
               template: (name, device: DataTableRowData) => {
-                const battery = device.battery_entity
-                  ? this.hass.states[device.battery_entity]
-                  : undefined;
-                // Have to work on a nice layout for mobile
                 return html`
-                  ${name}<br />
-                  ${device.area} | ${device.integration}<br />
-                  ${battery && !isNaN(battery.state as any)
-                    ? html`
-                        ${battery.state}%
-                        <ha-state-icon
-                          .hass=${this.hass!}
-                          .stateObj=${battery}
-                        ></ha-state-icon>
-                      `
-                    : ""}
+                  ${name}
+                  <div class="secondary">
+                    ${device.area} | ${device.integration}
+                  </div>
                 `;
               },
             },
+            battery_entity: {
+              title: this.hass.localize(
+                "ui.panel.config.devices.data_table.battery"
+              ),
+              sortable: true,
+              type: "numeric",
+              width: "90px",
+              template: (batteryEntity: string) => {
+                const battery = batteryEntity
+                  ? this.hass.states[batteryEntity]
+                  : undefined;
+                return battery && !isNaN(battery.state as any)
```

Comment:
If `isNaN`, we should still render a battery icon, just render it with a question mark.